### PR TITLE
Update requirements to match test results

### PR DESCRIPTION
**Pull request by tpelser**
We have conducted tests for various dependencies and update the requirements and requirements-dev files to match these, as is on the conda-forge feedstock.

### CODE REVIEW

File: requirements.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ dependencies:
...
   - conda-forge::descartes
   - conda-forge::pandas
-  - conda-forge::numpy<1.24
+  - conda-forge::numpy<=1.24
   - conda-forge::scipy
   - conda-forge::scikit-learn
-  - conda-forge::geokit>=1.4.0
-  - conda-forge::gdal=3.4.*
+  - conda-forge::gdal>2.4,<=2.4.4
+  - conda-forge::geokit=1.3.0
```
Comment (chrisjwin) [inline requirements.yml:13]: Is it necessary to fix geokit 1.3.0 exactly or could this be changed to >=1.3.0? @tpelser

**Review by chrisjwin — COMMENTED** [review]
I trust the versions that you suggested were tested. Only a minor remark concerning the geokit version @tpelser

File: requirements.yml (hunk 1 above)
Comment (tpelser) [reply requirements.yml:13]: yes this is necessary due to the current GDAL constraints

Comment (chrisjwin) [reply requirements.yml:13]: Perfect, just wanted to confirm. We must then check every time, however, if we can release the constrain when newer gdal/geokit versions come out.

**Review by chrisjwin — APPROVED** [review]

**Pull request merged**